src/FakeItEasy/ArgumentValidationConfigurationExtensions.cs

```diff
@@ -20,5 +26,127 @@ public static TInterface WithAnyArguments<TInterface>(this IArgumentValidationCo
 
             return configuration.WhenArgumentsMatch(x => true);
         }
+
+        /// <summary>
+        /// Configures the call to be accepted when the specified predicate returns true.
+        /// </summary>
+        /// <typeparam name="TInterface">The type of the interface.</typeparam>
+        /// <typeparam name="T1">The type of the first argument.</typeparam>
+        /// <param name="configuration">The configuration.</param>
+        /// <param name="argumentsPredicate">The argument predicate.</param>
+        /// <returns>A configuration object.</returns>
+        public static TInterface WhenArgumentsMatch<TInterface, T1>(
+            this IArgumentValidationConfiguration<TInterface> configuration,
+            Func<T1, bool> argumentsPredicate)

```

Comment:
Seems redundant, as one could just have `A<T>.That.Matches(…)`, but I guess it would look funny without it.